src/Functions/formatDateTime.cpp

```diff
@@ -943,7 +943,15 @@ class FunctionFormatDateTimeImpl : public IFunction
         {
             if constexpr (std::is_same_v<DataType, DataTypeDateTime64>)
             {
-                const auto c = DecimalUtils::split(vec[i], scale);
```

Comment:
It looks like for datetime64 needs a separate split function returning a second/fraction.